### ISSUE TITLE
Spack blame output incorrect for blank keys

### ISSUE THREAD

**Issue by sethrj** [LLNL/spack#22866]
As I mentioned in https://github.com/spack/spack/pull/22598#issuecomment-808896029 , the "empty" keys in the default modules.yaml file aren't rendered correctly.

### Steps to reproduce the issue

```console
$ spack config blame modules
# ...snip...
etc/spack/defaults/modules.yaml:32             share/pkgconfig:
etc/spack/defaults/modules.yaml:33             - PKG_CONFIG_PATH
                                               ? ''
                                               : - CMAKE_PREFIX_PATH
```

Prior to #22598 the blank lines had the correct line numbers corresponding to the values in `defaults/modules.yaml`:
```
34:    '':
35:      - CMAKE_PREFIX_PATH
```

Expected output:

```console
$ spack config blame modules
# ...snip...
etc/spack/defaults/modules.yaml:32             share/pkgconfig:
etc/spack/defaults/modules.yaml:33             - PKG_CONFIG_PATH
etc/spack/defaults/modules.yaml:34             '':
etc/spack/defaults/modules.yaml:35             - CMAKE_PREFIX_PATH
```
### Information on your system

* **Spack:** 0.16.1-2140-19b6d3589a
* **Python:** 3.8.5
* **Platform:** darwin-bigsur-skylake
* **Concretizer:** original

**Comment by alalazo** [comment]
Bug still present on

* **Spack:** 0.17.1-735-f8d4b4746c
* **Python:** 3.9.7
* **Platform:** linux-ubuntu20.04-icelake
* **Concretizer:** clingo

**Comment by sethrj** [comment]
@tgamblin @becker33 You may want to fix this before 0.18 is released, because with the warning message
```
==> Warning: Top-level properties "enable", "prefix_inspections", "lmod" in module config are ignored as of Spack v0.18. They should be set on the "default" module set. Run

	$ spack config update modules

to update the file to the new format
```

the modules file
```yaml
modules:
  enable::
    - lmod
  prefix_inspections:
    '':
    - CMAKE_PREFIX_PATH
    - '${PACKAGE}_ROOT'
```
creates a ~~broken~~ very odd `modules.yaml` file (though thankfully it leaves the backup):
```yaml
modules:
  prefix_inspections:
    ? ''
    : - CMAKE_PREFIX_PATH
      - ${PACKAGE}_ROOT
```

EDIT: it looks this bizarre syntax is valid YAML, and it preserves the original behavior when regenerating the module files. So I guess it's not *too* broken.
